services/configuration_repository/configuration_repository_decorator.rb

```diff
@@ -0,0 +1,31 @@
+require_relative "../services"
+
+module FastlaneCI
+  # This module acts as a decorator to be added to any method that potentially will make changes
+  # into the ci-config repository directory.
+  module ConfigurationRepositoryUpdater
+    # This decorator method is responsible of pulling changes from the ci-config repo.
+    # rubocop:disable Style/ClassVars
+    @@mutex = Mutex.new
```

Comment:
Let's not do `@@`, instead we can do:
```
    class << self
      attr_accessor :mutex
    end
    ConfigurationRepositoryUpdater.mutex = Mutex.new
```

Reply:
Fell into the old-school Ruby syntax for clarity and less verbosity, but you're right.